src/app/client/src/app/modules/dashboard/components/list-all-reports/list-all-reports.component.ts

```diff
@@ -0,0 +1,147 @@
+})
+export class ListAllReportsComponent implements OnInit {
+
+  constructor(public resourceService: ResourceService, private reportService: ReportService, private activatedRoute: ActivatedRoute,
+    private router: Router) { }
+
+  public reportsList$: Observable<any>;
+  public noResultFoundError: string;
+  private _isUserReportAdmin: boolean;
+
+  ngOnInit() {
+    this.reportsList$ = this.reportService.isAuthenticated(_.get(this.activatedRoute, 'snapshot.data.roles')).pipe(
+      mergeMap((isAuthenticated: boolean) => {
+        return isAuthenticated ? this.reportService.isUserReportAdmin().pipe(
+          tap(isReportAdmin => this._isUserReportAdmin = isReportAdmin),
+          mergeMap(isReportAdmin => this.getReportsList(isReportAdmin))
+        ) : throwError({ messageText: 'messages.stmsg.m0144' });
+      }),
+      catchError(err => {
+        this.noResultFoundError = _.get(err, 'messageText') || 'messages.stmsg.m0006';
+        return of({
+          table: {},
+          count: 0
+        });
+      })
+    );
+  }
+
+  /**s
+   * @private
+   * @returns Observable with list of reports.
+   * @memberof ListAllReportsComponent
+   */
+  private getReportsList(isUserReportAdmin: boolean) {
+    const filters = {
+      status: ['live', ...isUserReportAdmin ? ['draft', 'retired'] : []]
+    };
+    return this.reportService.listAllReports(filters).pipe(
+      map((apiResponse: { reports: any[], count: number }) => {
+        const reports = _.map(apiResponse.reports, report => {
+          return {
+            ..._.pick(report, ['reportid', 'title', 'description', 'reportgenerateddate',
+              'tags', 'updatefrequency'], ...isUserReportAdmin ? ['status'] : [])
+          };
+        });
+        const headers = ['reportid', 'Title', 'Description', 'Last Updated Date', 'Tags', 'Update Frequency',
+          ...isUserReportAdmin ? ['Status'] : []];
+        const result = {
+          table: {
+            header: headers || _.keys(reports[0]),
+            data: _.map(reports, report => _.values(report)),
+            defs: this.getColumnsDefs(),
+            options: {
+              searching: true
+            }
+          },
+          count: _.get(apiResponse, 'count')
+        };
+        return result;
+      })
+    );
+  }
+
+  /**
+   * @description returns columns defs to configure datatable.net
+   * @private
+   * @memberof ListAllReportsComponent
+   */
+  private getColumnsDefs() {
+    return [
+      {
+        targets: 0,
+        visible: false
+      },
+      {
+        targets: [1, 2],
+        width: '25%'
+      },
+      {
+        targets: [4],
+        width: '15%',
+        render: data => {
+          if (Array.isArray(data)) {
+            return _.map(data, tag => `<div class="sb-label sb-label-table sb-label-primary-100">${_.startCase(_.toLower(tag))}</div>`);
+          }
+          return _.startCase(_.toLower(data));
+        }
+      },
+      {
+        targets: [3, 5, 1, 2],
```

Comment:
This will change over the next weeks.  Once the schema is finalized and what needs to be shown in the table depending upon the roles. It will change later. 
Moreover data table component needs to be made configurable/dynamic to behave as per the column type. That we can do in some separate tech tasks.